backend/routes/signUpAuth.ts

```diff
@@ -0,0 +1,47 @@
+import * as dotenv from 'dotenv';
+import path from 'path';
+dotenv.config({path: path.resolve(__dirname, "../.env")});
+import app from '../utils/app';
+import cors from 'cors';
+import prisma from '../utils/db';
+import { PrismaClient } from '@prisma/client';
+import morgan from 'morgan';
+import express from 'express';
+
+import passport from 'passport';
+import {Strategy as LocalStrategy} from 'passport-local';
+import crypto from 'crypto'
+import mysql from 'mysql2';
+
+const db = prisma;
+
+app.post('/signup', async function(req, res, next) {
+    console.log(req.body);
+    console.log("hello");
+    const salt = crypto.randomBytes(16).toString("base64");
+    crypto.pbkdf2(req.body.password, salt, 310000, 32, 'sha256', async function(err, hashedPassword) { 
+        if(err) {
+            next(err); 
+        }
+        const user = await db.user.create({
+            data: {
+                email: req.body.email,
+                firstName: req.body.firstName,
+                lastName: req.body.lastName, 
+                password: hashedPassword.toString("base64"),
+                salt: salt //fix salt ??? how to best generate salt using crypto 
```

Comment:
you already generated it correctly on line 21!

also a cool javascript shorthand is whenever you have an object with fields like this:

```
{
  salt: salt,
}
```

it can be rewritten as

```
{
  salt,
}
```

Reply:
see: https://javascript.plainenglish.io/object-literals-using-object-property-shorthand-6360825c60ef#:~:text=When%20ES6%20was%20created%20object,the%20name%20and%20the%20key.&text=myKeyName-,%7D%3B,us%20to%20write%20less%20code!